packages/client/src/components/CommentVoteButtons.tsx

```diff
@@ -0,0 +1,109 @@
+import { HStack, IconButton, useColorModeValue, VStack } from '@chakra-ui/react';
+import React from 'react';
+import { MdArrowCircleDown, MdArrowCircleUp } from 'react-icons/md';
+import { useAuthContext } from '../context/AuthContext';
+import {
+  useAddCommentVote,
+  useDeleteCommentVote,
+  useUpdateCommentVote,
+} from '../hooks/useCommentVotes';
+import { useAddPostVote, useDeletePostVote, useUpdatePostVote } from '../hooks/usePostVote';
+import { VoteCount } from './VoteCount';
+
+interface Props {
+  postId: string;
+  parentId: string;
+  commentId: string;
+  voteStatus: boolean | null;
+  upvotes: string;
+  downvotes: string;
+}
+
+export const CommentVoteButtons = (props: Props) => {
+  const { authToken } = useAuthContext();
+  const addVoteMutation = useAddCommentVote(props.parentId, props.postId);
+  const updateVoteMutation = useUpdateCommentVote(props.parentId, props.postId);
+  const deleteVoteMutation = useDeleteCommentVote(props.parentId, props.postId);
+
+  const voteButtonHoverBg = useColorModeValue('gray.100', 'gray.600');
+  const upvoteButtonColorScheme = useColorModeValue('green', 'teal');
+  const downvoteButtonColorScheme = useColorModeValue('red', 'red');
+
+  const onUpvote = React.useCallback(() => {
+    if (authToken && props.voteStatus == null) {
+      const request = {
+        commentId: props.commentId,
+        vote: true,
+      };
+
+      return addVoteMutation.mutateAsync({ ...request, token: authToken }).then((res) => {});
```

Comment:
also if you're not doing anything with the result, just calling `.mutateAsync` you shouldn't have to call the `.then((res) => {})`, should be able to omit that.